### PR TITLE
Refactor/order cart

### PR DESCRIPTION
**Pull request by ganeshkumar1989**
| Question         | Answer
| ---------------- | ---
| Branch?          | `develop`
| Bug fix?         | yes
| New feature?     | no
| Breaking change? | no
| Tickets          | Fix #MANAGER-6472
| License          | BSD 3-Clause

## Description

Move order-cart.service.js to a new module - `order`. Reuse this module in web, pci and everywhere else.

### CODE REVIEW

File: packages/manager/modules/order/src/cart/order-cart.service.js [hunk 1]
```diff
@@ -93,19 +93,19 @@ export default class OrderCartService {
...
-  getProductOptions(cartId, productName, productPlanCode) {
+  getProductOptions(cartId, productName, planCode) {
     return this.OvhApiOrder.Cart()
       .Product()
       .v6()
       .getOptions({
         cartId,
         productName,
-        ...productPlanCode,
+        planCode,
```
Comment (mohammed-zahaf) [inline packages/manager/modules/order/src/cart/order-cart.service.js:108]: @ganeshkumar1989, we don't need to destruct it like for `productPlanCode` ?

Comment (ganeshkumar1989) [reply packages/manager/modules/order/src/cart/order-cart.service.js:108]: This method is being called with only the plan code and it does not need to be an object. Hence, we are now calling this method with the plancode.

File: packages/manager/modules/order/src/cart/order-cart.service.js [hunk 2]
```diff
@@ -212,7 +226,7 @@ export default class OrderCartService {
       ).$promise;
 
     if (
-      !checkout.autoPayWithPreferredPaymentMethod &&
+      !(checkout && checkout.autoPayWithPreferredPaymentMethod) &&
```
Comment (JeremyDec) [inline packages/manager/modules/order/src/cart/order-cart.service.js:229]: Maybe
```suggestion
      !checkout?.autoPayWithPreferredPaymentMethod &&
```

**Review by JeremyDec — APPROVED** [review]
LGTM

**Review by marie-j — DISMISSED** [review]

File: packages/manager/modules/order/src/cart/order-cart.service.js (hunk 2 above)
Comment (ganeshkumar1989) [reply packages/manager/modules/order/src/cart/order-cart.service.js]: This is done. Thanks

**Review by JeremyDec — DISMISSED** [review]

**Review by Jisay — DISMISSED** [review]

**Comment by JeremyDec** [comment]
@ganeshkumar1989 the build have its **test** workflow failed, can you check please ?

**Comment by ganeshkumar1989** [comment]
> @ganeshkumar1989 the build have its **test** workflow failed, can you check please ?

This is fixed. Thanks!

**Review by JeremyDec — APPROVED** [review]

**Review by JeremyDec — APPROVED** [review]

**Review by antleblanc — CHANGES_REQUESTED** [review]
Changes looks good to me.

Could you please describe how to use this new module in the `README.md` file?
see: `packages/manager/modules/order/README.md`

**Comment by antleblanc** [comment]
@ganeshkumar1989 Could you please resolve existing conflicts?
Thank you!

**Comment by antleblanc** [comment]
Hi @ganeshkumar1989!

Could you please resolve existing conflicts? In order to re-request and quality check and then be able to merge it.
Thank you!

**Review by JeremyDec — APPROVED** [review]

**Comment by antleblanc** [comment]
Closing this Pull Request.

cc @radireddy

**Pull request closed without merging**